File: models/order.js
const { pool } = require('../config/database');
const { v4: uuidv4 } = require('uuid');

class Order {
  static async create(orderData) {
    const connection = await pool.getConnection();
    
    try {
      await connection.beginTransaction();
      
      const orderId = uuidv4();
      const { customerId, total, items } = orderData;
      
      // Create order
      await connection.execute(
        'INSERT INTO orders (id, customer_id, total) VALUES (?, ?, ?)',
        [orderId, customerId, total]
      );
      
      // Create order items
      for (const item of items) {
        const itemId = uuidv4();
        await connection.execute(
          'INSERT INTO order_items (id, order_id, product_id, product_name, price, quantity, subtotal) VALUES (?, ?, ?, ?, ?, ?, ?)',
          [itemId, orderId, item.productId, item.productName, item.price, item.quantity, item.subtotal]
        );
        
        // Update product stock
        await connection.execute(
          'UPDATE products SET stock = stock - ? WHERE id = ?',
          [item.quantity, item.productId]
        );
      }
      
      await connection.commit();
      return await this.getById(orderId);
    } catch (error) {
      await connection.rollback();
      throw error;
    } finally {
      connection.release();
    }
  }

  static async getById(id) {
    try {
      const [orderRows] = await pool.execute(`
        SELECT o.*, c.name as customer_name, c.email as customer_email, c.address as customer_address
        FROM orders o
        LEFT JOIN customers c ON o.customer_id = c.id
        WHERE o.id = ?
      `, [id]);
      
      if (orderRows.length === 0) return null;
      
      const [itemRows] = await pool.execute(`
        SELECT * FROM order_items WHERE order_id = ?
      `, [id]);
      
      return {
        ...orderRows[0],
        items: itemRows
      };
    } catch (error) {
      throw error;
    }
  }

  static async getAll() {
    try {
      const [rows] = await pool.execute(`
        SELECT o.*, c.name as customer_name, c.email as customer_email
        FROM orders o
        LEFT JOIN customers c ON o.customer_id = c.id
        ORDER BY o.created_at DESC
      `);
      return rows;
    } catch (error) {
      throw error;
    }
  }
}

module.exports = Order;

File: models/order.py
import uuid
from datetime import datetime
from config.database import get_storage, data_lock

class Order:
    @staticmethod
    def create(order_data):
        """Create a new order"""
        try:
            with data_lock:
                storage = get_storage()

                order_id = str(uuid.uuid4())
                customer_id = order_data['customerId']
                total = order_data['total']
                items = order_data['items']

                # Create order
                order = {
                    'id': order_id,
                    'customer_id': customer_id,
                    'total': float(total),
                    'status': 'pending',
                    'created_at': datetime.now().isoformat()
                }
                storage['orders'][order_id] = order

                # Create order items and update stock
                for item in items:
                    item_id = str(uuid.uuid4())
                    order_item = {
                        'id': item_id,
                        'order_id': order_id,
                        'product_id': item['productId'],
                        'product_name': item['productName'],
                        'price': float(item['price']),
                        'quantity': item['quantity'],
                        'subtotal': float(item['subtotal'])
                    }
                    storage['order_items'][item_id] = order_item

                    # Update product stock
                    if item['productId'] in storage['products']:
                        storage['products'][item['productId']]['stock'] -= item['quantity']

            return Order.get_by_id(order_id)
        except Exception as error:
            raise error

    @staticmethod
    def get_by_id(order_id):
        """Get an order by ID with its items"""
        try:
            with data_lock:
                storage = get_storage()

                if order_id not in storage['orders']:
                    return None

                order = storage['orders'][order_id].copy()

                # Get customer info
                customer_id = order.get('customer_id')
                if customer_id and customer_id in storage['customers']:
                    customer = storage['customers'][customer_id]
                    order['customer_name'] = customer['name']
                    order['customer_email'] = customer['email']
                    order['customer_address'] = customer['address']

                # Get order items
                items = []
                for item_id, item in storage['order_items'].items():
                    if item['order_id'] == order_id:
                        items.append(item.copy())

                order['items'] = items

                return order
        except Exception as error:
            raise error

    @staticmethod
    def get_all():
        """Get all orders with customer info"""
        try:
            with data_lock:
                storage = get_storage()

                orders = []
                for order_id, order in storage['orders'].items():
                    order_copy = order.copy()

                    # Get customer info
                    customer_id = order_copy.get('customer_id')
                    if customer_id and customer_id in storage['customers']:
                        customer = storage['customers'][customer_id]
                        order_copy['customer_name'] = customer['name']
                        order_copy['customer_email'] = customer['email']

                    orders.append(order_copy)

                # Sort by created_at descending
                orders.sort(key=lambda x: x.get('created_at', ''), reverse=True)

                return orders
        except Exception as error:
            raise error
